network.py

```diff
@@ -83,8 +127,14 @@ def accuracy(y_true, y_pred):
     return K.mean(K.equal(y_true, K.cast(y_pred < 0.5, y_true.dtype)))
 
 
-def compile(model):
-    opt = RMSprop()
-    # opt = SGD(lr=0.0003, decay=1e-6, momentum=0.9, nesterov=True)
-    # opt = Nadam()
-    model.compile(loss=contrastive_loss, optimizer=opt, metrics=[accuracy])
+def compile(model, optimizer=None, loss_func=False):
+
+    allOptimizers = {'sgd': SGD(lr=0.0003, decay=1e-6, momentum=0.9, nesterov=True),
+        'adam': Adam(), 'nadam': Nadam(), 'rms': RMSprop()}
+
+    if optimizer in allOptimizers:
```

Comment:
Let's turn this into an `assert`, so we ensure we always pass a valid optimizer.

Reply:
sorry I missed this comment. Apologies!